doc/development/glosary.md

```diff
@@ -0,0 +1,28 @@
+# Puter's glosary
+
+## General
+
+1. **Puter:** Puter is an advanced, open-source internet operating system designed to be feature-rich, exceptionally fast, and highly extensible.
+
+2. **Self hosting:** Means running and managing the app on your infrastructure instead of using thid-party hosting service.
+You can learn more on how to self host your Puter instance [here](./deployment/self_host.md)
+
+3. **Kernel:** It is a simple module which is in charge of orchestrating the initialization of the system, following the boot sequence. Following the [Microkernel pattern](https://www.oreilly.com/library/view/software-architecture-patterns/9781098134280/ch04.html), this module corresponds to the Core of Puter backend.
```

Comment:
Since Puter's kernel runs "modules", it might be confusing to call the kernel itself a module.

```suggestion
3. **Kernel:** It is a simple class which is in charge of orchestrating the initialization of the system, following the boot sequence. Following the [Microkernel pattern](https://www.oreilly.com/library/view/software-architecture-patterns/9781098134280/ch04.html), this class corresponds to the Core of Puter backend.
```